src/awx_plugins/credentials/azure_kv.py

```diff
@@ -72,24 +94,72 @@
     ],
     'required': [
         'url',
-        'client',
-        'secret',
-        'tenant',
         'secret_field',
     ],
 }
 
 
-def azure_keyvault_backend(**kwargs):
-    csc = ClientSecretCredential(
-        tenant_id=kwargs['tenant'],
-        client_id=kwargs['client'],
-        client_secret=kwargs['secret'],
-    )
-    kv = SecretClient(credential=csc, vault_url=kwargs['url'])
-    return kv.get_secret(
-        name=kwargs['secret_field'],
-        version=kwargs.get('secret_version', ''),
+def _initialize_credential(
+    tenant: str = '',
+    client: str = '',
+    secret: str = '',
+) -> TokenCredential:
+    explicit_credentials_provided = all((tenant, client, secret))
+
+    if explicit_credentials_provided:
+        return ClientSecretCredential(
+            tenant_id=tenant,
+            client_id=client,
+            client_secret=secret,
+        )
+
+    try:
+        return ManagedIdentityCredential()
+    except CredentialUnavailableError as cred_err:
+        raise LookupError(
+            'Failed to detect credentials from environment.',
+        ) from cred_err
```

Comment:
This is not covered by tests, so we don't know if it actually works. It might not be.
https://app.codecov.io/gh/ansible/awx-plugins/commit/94f7e3611fd08fcec49e22d81d6f7618ab8567fc/blob/src/awx_plugins/credentials/azure_kv.py?flags%5B0%5D=pytest&dropdown=coverage#L118